test/Multigoal.lean

```diff
@@ -0,0 +1,105 @@
+import Mathlib.Tactic.Basic
+import Mathlib.Tactic.Conv
+import Mathlib.Tactic.Linter.Multigoal
+
+-- The warning generated by `linter.multiGoal` is not suppressed by `#guard_msgs`,
+-- because the linter is run on `#guard_msgs` itself. This is a known issue, see e.g.
+-- https://leanprover.zulipchat.com/#narrow/stream/348111-batteries/topic/unreachableTactic.20linter.20not.20suppressed.20by.20.60.23guard_msgs.60
+-- We jump through an extra hoop here to silence the warning.
+set_option linter.multiGoal false
+
+-- A deactivated linter does nothing.
+example : True := by
+  by_cases 0 = 0
+  exact .intro
+  exact .intro
+
+#guard_msgs(drop warning) in
+set_option linter.multiGoal true in
+/--
+warning: 'exact .intro' leaves 1 goal 'Lean.Parser.Tactic.exact'
+note: this linter can be disabled with `set_option linter.multiGoal false`
+-/
+#guard_msgs in
+example : True := by
+  by_cases 0 = 0
+  exact .intro
+  exact .intro
+
+#guard_msgs(drop warning) in
+set_option linter.multiGoal true in
+/--
+warning: 'assumption' leaves 1 goal 'Lean.Parser.Tactic.assumption'
+note: this linter can be disabled with `set_option linter.multiGoal false`
+-/
+-- the linter keeps linting after ignoring a `conv`, `conv_lhs`, `conv_rhs`
+#guard_msgs in
+example {n : Nat} (hn : n = 0) : n + 0 = 0 := by
+  conv =>
+    congr
+    rw [← Nat.add_zero 0]
+  conv_lhs =>
+    congr
+    rw [← Nat.add_zero n]
+    rfl
+  conv_rhs =>
+    rw [← Nat.add_zero 0]
+    congr
+    rfl
+    rfl
+  by_cases 0 = 0
+  assumption
+  assumption
+
+#guard_msgs(drop warning) in
+set_option linter.multiGoal true in
+/--
+warning: 'rfl' leaves 1 goal 'Lean.Parser.Tactic.tacticRfl'
+note: this linter can be disabled with `set_option linter.multiGoal false`
+-/
+-- the linter allows `iterate` and `repeat'`, but continues to lint.
+#guard_msgs in
+example (p : Prop) (hp : p) : (0 = 0 ∧ p) ∨ 0 = 0 := by
+  iterate left; decide
+  repeat' left; decide
+  refine Or.inl ⟨?_, ?_⟩
+  rfl
+  assumption
+
+#guard_msgs(drop warning) in
+set_option linter.multiGoal true in
+/--
+warning: 'rfl' leaves 2 goals 'Lean.Parser.Tactic.tacticRfl'
```

Comment:
Added this test, to clarify which lines warn, and to show that several warnings also work.